license-header-check/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'License Header Check'
+description: 'check license/copyright header'
+inputs:
+  included_file_patterns:
+    description: "included file pattern"
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get changed files
+      shell: bash
+      run: |
+        CURRENT_YEAR=$(date +%Y)
+        LICENSE_PATTERN="Copyright \(c\) .*?${CURRENT_YEAR}, NVIDIA CORPORATION."
```

Comment:
Yes, in current rules, it will fail if the file is renamed only. We optimize in 2 options:

1. `git diff --name-only --diff-filter=R` ignore all renamed files.
2. `git diff --name-only --diff-filter=AM` detect new added and modified files. If the content of renamed file is modified, we still check the header.

Reply:
Updated the action with option 3. Renamed files with content changes will be checked.